src/PointwiseFunctions/GeneralRelativity/GeneralizedHarmonic/Christoffel.cpp

```diff
@@ -44,6 +44,81 @@ auto christoffel_second_kind(
   christoffel_second_kind(make_not_null(&christoffel), phi, inv_metric);
   return christoffel;
 }
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+tnsr::a<DataType, SpatialDim, Frame> trace_christoffel(
+    const tnsr::a<DataType, SpatialDim, Frame>& spacetime_normal_one_form,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_normal_vector,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) {
+  auto trace = make_with_value<tnsr::a<DataType, SpatialDim, Frame>>(pi, 0.0);
+  trace_christoffel<SpatialDim, Frame, DataType>(
+      &trace, spacetime_normal_one_form, spacetime_normal_vector,
+      inverse_spatial_metric, inverse_spacetime_metric, pi, phi);
+  return trace;
+}
+
+template <size_t SpatialDim, typename Frame, typename DataType>
+void trace_christoffel(
+    const gsl::not_null<tnsr::a<DataType, SpatialDim, Frame>*> trace,
+    const tnsr::a<DataType, SpatialDim, Frame>& spacetime_normal_one_form,
+    const tnsr::A<DataType, SpatialDim, Frame>& spacetime_normal_vector,
+    const tnsr::II<DataType, SpatialDim, Frame>& inverse_spatial_metric,
+    const tnsr::AA<DataType, SpatialDim, Frame>& inverse_spacetime_metric,
+    const tnsr::aa<DataType, SpatialDim, Frame>& pi,
+    const tnsr::iaa<DataType, SpatialDim, Frame>& phi) {
+  destructive_resize_components(trace,
+                                get_size(get<0>(spacetime_normal_one_form)));
+  get<0>(*trace) = 0.0;
+  // Compute common terms between components.
+  for (size_t b = 0; b < SpatialDim + 1; ++b) {
```

Comment:
If we want to start using tensor expressions in more places, maybe consider changing this to use tensor expressions? Up to you though!

Reply:
Yea, I thought about that but want to just keep this as-is for now. We need to do a full performance audit on GH anyway and should explore TEs then

Reply:
Ok, works for me!

Reply:
I'm also OK doing for loops for this one